### DISCUSSION TITLE
How to access nested metadata having a key with dots

### DISCUSSION THREAD

**Discussion by vlcinsky** [quarto-dev/quarto-cli#10914] [Q&A] [answered]
### Description

My sample index.qmd:

```
---
title: My Page
nested:
  a.b.c:
    key: Show me
---

I would like to show the "Show me" using a shortcode.

E.g. {{< meta nested["a.b.c"]key >}}

but it fails.
```

My question is: is there a **syntax for accessing keys, which include dot(s) in their name**?

**Comment by mcanouil** [comment]
The issue is not about access to nested metadata but the fact that you are using dots in metadata keys which seems to be unsupported (kind of for obvious reasons) and that the brackets method is not supported in the shortcode. (cc @cscheid)

````qmd
---
a.b.c: Show me
format: html
---

{{< meta a.b.c >}}
````

For this particular shortcode, the code is the following:

- https://github.com/quarto-dev/quarto-cli/blob/acddc3e4068687cb93d647280e9b0b1aa5084b3a/src/resources/filters/quarto-pre/shortcodes-handlers.lua#L169-L201

**Reply by cscheid** [reply, marked as answer]
I created an issue to track https://github.com/quarto-dev/quarto-cli/issues/10936

**Reply by mcanouil** [reply]
@vlcinsky in the next pre-release version you will be able to access nested metadata data with dots by escaping them.
See https://github.com/quarto-dev/quarto-cli/issues/10936#issuecomment-2384089196
